## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454808`
Commit: [`266856b7e02f9b58a25705276c1463c5ca38c7c4`](https://github.com/NixOS/nixpkgs/commit/266856b7e02f9b58a25705276c1463c5ca38c7c4) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/266856b7e02f9b58a25705276c1463c5ca38c7c4..pull/454808/head))
Merge: [`bd8d83a269b4c5ec82046262f53b8a8f7b58878e`](https://github.com/NixOS/nixpkgs/commit/bd8d83a269b4c5ec82046262f53b8a8f7b58878e)

Logs: https://github.com/nartsisss/nixpkgs-review-gha/actions/runs/18742702071


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>splitcode</li>
  </ul>
</details>

---
### `aarch64-linux`
:white_check_mark: *No rebuilds*

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>splitcode</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>splitcode</summary>
<pre>  114 | #  define OS_CODE 19
      |           ^
/nix/build/nix-11123-3606565695/source/ext/zlib-ng/arch/x86/../../zutil.h:102:11: note: previous definition is here
  102 | #  define OS_CODE  7
      |           ^
clang: warning: argument unused during compilation: &#x27;-fno-semantic-interposition&#x27; [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: &#x27;-fno-semantic-interposition&#x27; [-Wunused-command-line-argument]
1 warning generated.
[ 96%] Building C object CMakeFiles/zlib.dir/gzlib.c.o
[ 97%] Building C object CMakeFiles/zlib.dir/gzread.c.o
[ 98%] Building C object CMakeFiles/zlib.dir/gzwrite.c.o
clang: warning: argument unused during compilation: &#x27;-fno-semantic-interposition&#x27; [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: &#x27;-fno-semantic-interposition&#x27; [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: &#x27;-fno-semantic-interposition&#x27; [-Wunused-command-line-argument]
[100%] Linking C shared library libz.dylib
[100%] Built target zlib
[ 90%] No install step for &#x27;zlib-ng&#x27;
[ 95%] Completed &#x27;zlib-ng&#x27;
[ 95%] Built target zlib-ng
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>splitcode</li>
  </ul>
</details>

--- a/pkgs/by-name/sp/splitcode/package.nix
+++ b/pkgs/by-name/sp/splitcode/package.nix
@@ -7,18 +7,15 @@
   zlib,
   nix-update-script,
 }:
-let
-  version = "0.31.3";
-in
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "splitcode";
-  inherit version;
+  version = "0.31.4";
 
   src = fetchFromGitHub {
     owner = "pachterlab";
     repo = "splitcode";
-    tag = "v${version}";
-    hash = "sha256-EEYjBo52jPCSnv5WSGsXhfZEjsBCHdGerVPOZfShXBU=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-jaZptrF5HSQSe2KQdBNPprhsVZ2FSudZFgWPn9zQq3A=";
   };
 
   nativeBuildInputs = [ cmake ];
@@ -27,6 +24,12 @@ stdenv.mkDerivation {
     libcxx
     zlib
   ];
+
+  postPatch = ''
+    # https://github.com/pachterlab/splitcode/pull/46
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 2.8.12 3.10
+  '';
 
   doCheck = true;
   checkPhase = ''
@@ -49,4 +52,4 @@ stdenv.mkDerivation {
       "aarch64-linux"
     ];
   };
-}
+})

